packages/react-hooks/src/hooks/interval.ts

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+export function useInterval(callback: () => void, delay: number) {
+  React.useEffect(() => {
+    const id = setInterval(callback, delay);
+    return () => clearInterval(id);
```

Comment:
React clearing out the interval during rerenders will cause surprising behaviour.  e.g., a user editing form values will trigger a rerender, and the interval will be reset.

Here's a code sandbox to demonstrate the oddness: https://codesandbox.io/s/priceless-cdn-hlzc2  For as long as you type in the sandbox's input field, the interval won't fire.